### PR TITLE
generate wallet

### PR DESCRIPTION
**Pull request by devwistkey**
@tralkan

### CODE REVIEW

File: lib/blockchain/wallet.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.*/
+
+import Arweave from "arweave"
+import * as zcrypto from "@zilliqa-js/zilliqa"
```
Comment (txalkan) [inline lib/blockchain/wallet.ts]: import * as zcrypto from "@zilliqa-js/crypto"

**Comment by txalkan** [comment]
Hi @DevPH-Wistkey, let's move the src/lib files from the SSI Browser to tyron.js as well (and migrate to TypeScript). We have to remove "@zilliqa-js/zilliqa" & "arweave" from the SSI Browser.

**Comment by devwistkey** [comment]
Updated @tralkan

File: lib/blockchain/wallet.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.*/
+
+import Arweave from 'arweave'
+import * as zcrypto from '@zilliqa-js/zilliqa'
```
Comment (txalkan) [inline lib/blockchain/wallet.ts]: zilliqa-js/crypto

File: lib/blockchain/util.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        const hash_ = Array()
+        for (const element of elements) {
+            const h = hash.sha256().update(element).digest('hex')
+            hash_.push('0x' + h)
+            h_ = h_ + h
+        }
+        return [hash_, h_]
+    }
+
+    public static async AddLiquidity(
```
Comment (txalkan) [inline lib/blockchain/util.ts]: move AddLiquidity to /blockchain/defi-didDomain.ts

File: lib/blockchain/util.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        const tyron_: TransitionParams = {
+            vname: 'tyron',
+            type: 'Option Uint128',
+            value: tyron,
+        }
+        params.push(tyron_)
+        return params
+    }
+
+    public static async RemoveLiquidity(
```
Comment (txalkan) [inline lib/blockchain/util.ts]: move RemoveLiquidity to blockchain/defi-didDomain.ts

**Review by txalkan — CHANGES_REQUESTED** [review]

File: lib/blockchain/wallet.ts (hunk 2 above)
Comment (devwistkey) [reply lib/blockchain/wallet.ts]: done @tralkan

File: lib/blockchain/util.ts (hunk 3 above)
Comment (devwistkey) [reply lib/blockchain/util.ts]: done @tralkan

File: lib/blockchain/util.ts (hunk 4 above)
Comment (devwistkey) [reply lib/blockchain/util.ts]: done @tralkan

**Pull request merged**